cpp/src/parquet/properties.h

```diff
@@ -79,10 +93,45 @@ class PARQUET_EXPORT ReaderProperties {
   /// Disable buffered stream reading.
   void disable_buffered_stream() { buffered_stream_enabled_ = false; }
 
-  /// Return the size of the buffered stream buffer.
-  int64_t buffer_size() const { return buffer_size_; }
-  /// Set the size of the buffered stream buffer in bytes.
-  void set_buffer_size(int64_t size) { buffer_size_ = size; }
+  /// Return the default size of the buffered stream buffer.
+  int64_t buffer_size() const { return default_column_reader_properties_.buffer_size(); }
+  /// Set the default size of the buffered stream buffer in bytes.
+  void set_buffer_size(int64_t size) {
+    default_column_reader_properties_.set_buffer_size(size);
+  }
+
+  /// Return the size of the buffered stream buffer for a column chunk.
+  int64_t buffer_size(int row_group_index, int column_index) const {
```

Comment:
The current change looks better than leaking the property thing into the reader api. However, I'm not sure if it is appropriate to introduce concept like `row_group_index` into ReaderProperties, especially when `ColumnReaderPropertiesMap` is organized into a two-layer design. Sorry but I would -1 on making such change on optimization like this.

WDYT? @pitrou @emkornfield @mapleFU

Reply:
Personally I think deciding a buffer-size for per-row-group and per-column is to hack... Perhaps user here has define a `CalculateBufferSize(const ColumnChunkMetadata&)`, and caculating this.

it's ok but I guess it's too complex... Why a consistent buffer_size for each column is not ok? cc @pitrou